File: ios/Models/ImageDistortionResult.swift
//
//  ImageDistortionResult.swift
//  SeniorDesign
//
//  Created by Nicholas Chu on 2/28/22.
//

import Foundation

class ImageDistortionResult: NSObject, Encodable {
  
  private var globalQuality: Float;
  private var blurry: Float;
  private var shaky: Float;
  private var bright: Float;
  private var dark: Float;
  private var grainy: Float;
  private var none: Float;
  private var other: Float;
  
  init(globalQuality: Float, blurry: Float, shaky: Float, bright: Float, dark: Float, grainy: Float, none: Float, other: Float) {
    self.globalQuality = globalQuality;
    self.blurry = blurry;
    self.shaky = shaky;
    self.bright = bright;
    self.dark = dark;
    self.grainy = grainy;
    self.none = none;
    self.other = other;
  }
  
  public override var description: String {
    return "ImageDistortionResult(globalQuality=\(globalQuality), blurry=\(blurry), shaky=\(shaky), bright=\(bright), dark=\(dark), grainy=\(grainy), none=\(none), other=\(other))"
  }
  
  public func toJson() -> NSDictionary {
    do {
      let encodedData = try JSONEncoder().encode(self)
      let json = try JSONSerialization.jsonObject(with: encodedData, options: [])
      
      guard
        let dictionary = json as? [String : Any]
      else {
        return [:]
      }
      
      return dictionary as NSDictionary
    } catch {
      NSLog("Error converting ImageDistortionResult to dictionary!")
      return [:]
    }
  }
}

File: android/app/src/main/java/com/seniordesign/ImageDistortionResult.java
package com.seniordesign;

import com.facebook.react.bridge.WritableNativeMap;

public class ImageDistortionResult {
    private float globalQuality;
    private float blurry;
    private float shaky;
    private float bright;
    private float dark;
    private float grainy;
    private float none;
    private float other;

    public ImageDistortionResult(float globalQuality, float blurry, float shaky, float bright, float dark, float grainy, float none, float other) {
        this.globalQuality = globalQuality;
        this.blurry = blurry;
        this.shaky = shaky;
        this.bright = bright;
        this.dark = dark;
        this.grainy = grainy;
        this.none = none;
        this.other = other;
    }

    public static ImageDistortionResult empty() {
        return new ImageDistortionResult(0, 0, 0, 0, 0, 0, 0, 0);
    }

    public WritableNativeMap toJson() {
        WritableNativeMap map = new WritableNativeMap();
        map.putDouble("globalQuality", globalQuality);
        map.putDouble("blurry", blurry);
        map.putDouble("shaky", shaky);
        map.putDouble("bright", bright);
        map.putDouble("dark", dark);
        map.putDouble("grainy", grainy);
        map.putDouble("none", none);
        map.putDouble("other", other);
        return map;
    }
}
